apps/managedidentity/managedidentity.go

```diff
@@ -173,17 +280,104 @@ func createIMDSAuthRequest(ctx context.Context, id ID, resource string, claims s
 	return req, nil
 }
 
-func (client Client) getTokenForRequest(req *http.Request) (accesstokens.TokenResponse, error) {
+func createAzureArcAuthRequest(ctx context.Context, id ID, resource string, claims string) (*http.Request, error) {
+	identityEndpoint, _ := getAzureArcEnvironmentVariables()
+	var msiEndpoint *url.URL
+
+	if identityEndpoint == "" {
+		identityEndpoint = azureArcEndpoint
+	}
+
+	if _, ok := id.(systemAssignedValue); !ok {
+		return nil, errors.New("Azure Arc doesn't support specifying a user-assigned managed identity at runtime")
+	}
+
+	identityEndpoint = strings.Replace(identityEndpoint, "localhost", "127.0.0.1", -1)
+	msiEndpoint, parseErr := url.Parse(identityEndpoint)
+	if parseErr != nil {
+		return nil, fmt.Errorf("couldn't parse %q: %s", identityEndpoint, parseErr)
+	}
+
+	msiParameters := msiEndpoint.Query()
+	msiParameters.Set(apiVersionQueryParameterName, azureArcAPIVersion)
+	resource = strings.TrimSuffix(resource, "/.default")
+	msiParameters.Set(resourceQueryParameterName, resource)
+
+	if len(claims) > 0 {
+		msiParameters.Set("claims", claims)
+	}
+
+	msiEndpoint.RawQuery = msiParameters.Encode()
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, msiEndpoint.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating http request %s", err)
+	}
+	req.Header.Set(metaHTTPHeaderName, "true")
+	return req, nil
+}
+
+// GetEnvironmentVariables returns the identity and IMDS endpoints
+func getAzureArcEnvironmentVariables() (string, string) {
+	identityEndpoint := os.Getenv(IdentityEndpointEnvVar)
+	imdsEndpoint := os.Getenv(ArcIMDSEnvVar)
+
+	if identityEndpoint == "" || imdsEndpoint == "" {
+		platform := runtime.GOOS
+		fileDetectionPath, retrievedPath := azureArcFileDetection[platform]
+
+		if retrievedPath {
+			if _, err := os.Stat(fileDetectionPath); err == nil {
+				identityEndpoint = azureArcEndpoint
+				imdsEndpoint = himdsExecutableHelperString
```

Comment:
Why is this needed? Are you using this anywhere else after the env is determined?

Reply:
This is another thing I saw in MSAL JS: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/f0c5f0250741cfd673538ce51c19a345fdf9be75/lib/msal-node/src/client/ManagedIdentitySources/AzureArc.ts#L49

Reply:
The reason this is there in MSAL JS is because the flow is setup in a way where the env variables are confirmed again later in the flow to verify that it is arc env. I don't think you do that here. I think you can skip this.

Reply:
Ok perfect will update thanks